core/renderer/shaders/positionTextureGrayAlpha.frag

```diff
@@ -0,0 +1,16 @@
+#version 310 es
+precision highp float;
+precision highp int;
+
+layout(location = COLOR0) in vec4 v_color;
+layout(location = TEXCOORD0) in vec2 v_texCoord;
+
+layout(binding = 0) uniform sampler2D u_tex0;
+
+layout(location = SV_Target0) out vec4 FragColor;
+
+void main()
+{
+    vec4 c = texture(u_tex0, v_texCoord);
+    FragColor = v_color * vec4(c.r, c.r, c.r, c.g);
```

Comment:
This will fail on GLES2, because for two channel textures it uses `GL_LUMINANCE_ALPHA` pixel format which loads data into shader's rgba channels as LLLA. So for GLES2 you need to use `(c.r, c.r, c.r, c.a)`. So you either need an `#if defined(GLES2)`, or use compatibility macro `RG8_CHANNEL` defined in `base.glsl`. Check `videoTextureNV12.frag` shader for example.

Reply:
Would (c.r, c.r, c.r, c.a) work for both cases? For RG8, it's being loaded as RGRG into the RGBA channels, so instead of using R and G, why not just use R and A, since A has the same value as G?

Reply:
I treat loading RG8 as RGRG a bug, so I would go with an `#if defined(GLES2)`. GLES2 will die at some point (In few years? Currently just 4.1% of Android market share), so this R and A channels is a deprecated legacy, and I wouldn't build a new feature based on outdated stuff.